### PR TITLE
Implement json web tokens for authorization tokens

### LINKED ISSUE #76: use json web tokens instead of uuids for login token and magic links
this allows us to encode whatever extra meta data we want like email and expiration. this reduces the amount of information clients hand to lcs, and gives clients more info on login. it also allows other backends to verify that a token is valid since jwt's can be public key signed.

in the case of magic links this can also let us pass the email to the client.

### PR DESCRIPTION
**Pull request by DylanBelka**
Resolves #76 
Essentially replaces the token string in request bodies which used to be a randomly generated uuid with a stringified jwt that encodes user's email and an expiration timestamp.
Frontend and other consumers of LCS will need to be updated to properly take advantage of jwts. For now these applications will still be able to work without modification as they *should not* be actually using the token themselves for anything.

### CODE REVIEW

**Comment by Kironb** [comment]
We are going to put this on hold for merging until we set some other things up to ensure that nothing goes poorly once deployed since the JWT encodings may be different from the bcrypt encodings.

File: config.travis.py [hunk 1]
```diff
@@ -57,3 +57,6 @@ class TRAVEL:
 }
 RESUME_BUCKET = "resumet"
 
+# Json webtoken
+JWT_SECRET = "1002C8E19420D4D5F117C13BC1971875F7CE4791ED0901A60E46549AAA74C80F"
```
Comment (mjrb) [inline config.travis.py:61]: Make sure you are not using this same secret in Dev or prod. Kinda spooked me a bit seeing this. You can also use Travis envy vars to hide secrets like this

Comment (Kironb) [reply config.travis.py:61]: Yeah no worries, wasn't planning on setting on the same secret in dev or prod. Was just doing a quick fix for travis at the time.

**Comment by Kironb** [comment]
@mjrb For some clarification on what @DylanBelka said, since JWT's store information, encoded into their tokens we originally were sending redundant data (with the email and expiration date within the token as well as sent outside of it in the JSON). For example, we were sending
{ "auth":
             "token": jwt_token 
             "valid_until": time
              "email": email
}.

where jwt_token has:
{   
 "exp":time
 "email":email
} 

This was to ensure when we deployed there would not be an issue where frontend broke as a result. Since then, we have had a meeting with the frontend and we removed the redundant data and only have the expiration date and email encoded into the JWT. This could also potentially break other applications (such as mentorq, teambuilder, etc.) when it gets released.

So now we are just sending:
{"token": jwt_token}

File: tests/test_validate.py [hunk 2]
```diff
@@ -34,16 +38,23 @@ def test_validate_token():
     val = validate.validate({'email': user_email, 'oken': token + 'fl'}, None)
```
Comment (Anarav) [inline tests/test_validate.py:38]: Unrelated to this PR but a small bug fix that can be added alongside: 'oken' should be 'token'

File: tests/test_validate.py [hunk 3]
```diff
@@ -4,17 +4,21 @@
...
 @pytest.mark.run(order=3)
 def test_validate_token():
     user_email = "creep@radiohead.ed"
     passwd = "love"
     usr_dict = {'email': user_email, 'password': passwd}
     auth = authorize.authorize(usr_dict, None)
-    token = auth['body']['auth']['token']
+    token = auth['body']['token']
+
+    print("token is " + token)
```
Comment (Anarav) [inline tests/test_validate.py:21]: Debug leftover?

Comment (Kironb) [reply tests/test_validate.py:21]: Yeah, that looks like a debug leftover. We should probably actually add more print statements to our tests in general in the future so when something goes wrong we have a sort of log of what happened but that is mostly an unrelated goal.

**Pull request merged**